fog/overseer/server/src/worker.rs

```diff
@@ -235,12 +246,23 @@ where
     /// Returns the latest round of ingest summaries for each
     /// FogIngestGrpcClient that communicates with a node that is online.
     fn retrieve_ingest_summary_node_mappings(
-        &self,
+        &mut self,
     ) -> Result<Vec<IngestSummaryNodeMapping>, OverseerError> {
         let mut ingest_summary_node_mappings: Vec<IngestSummaryNodeMapping> = Vec::new();
         for (ingest_client_index, ingest_client) in self.ingest_clients.iter().enumerate() {
             match ingest_client.get_status() {
                 Ok(ingest_summary) => {
+                    if self
+                        .unresponsive_node_urls
+                        .contains(ingest_client.get_uri())
+                    {
+                        self.unresponsive_node_urls.remove(ingest_client.get_uri());
```

Comment:
Optional/Equivalent: Use the return value of `remove()`
```suggestion
                    if self
                        .unresponsive_node_urls
                        .remove(ingest_client.get_uri())
                    {
```